python/cuspatial/cuspatial/core/binpreds/feature_overlaps.py

```diff
@@ -36,6 +39,17 @@ class OverlapsPredicateBase(EqualsPredicateBase):
     pass
 
 
+class PolygonPolygonOverlaps(BinPred):
+    def _preprocess(self, lhs, rhs):
+        contains_lhs = lhs.contains(rhs)
+        contains_rhs = rhs.contains(lhs)
+        contains_properly_lhs = _basic_contains_properly_any(lhs, rhs)
+        contains_properly_rhs = _basic_contains_properly_any(rhs, lhs)
+        return ~(contains_lhs | contains_rhs) & (
+            contains_properly_lhs | contains_properly_rhs
+        )
```

Comment:
This seems a bit conflicting to me. Negating contains and then `contains_properly`?

Reply:
For polygons a and b to overlap, it is necessary that neither polygon completely contain the other polygon, but that some of the points in a or b be contained by the other. https://postgis.net/docs/ST_Overlaps.html. Let me know if this answer is suficient.

Reply:
To my understanding, the only difference between contains and contains_properly, are that if one point falls on the boundary of another, it does/does not counted as contains. The first part of the statement:

> neither polygon completely contain the other polygon

is correctly captured by `~(contains_lhs | contains_rhs)`.

However, this statement isn't:

> that some of the points in a or b be contained by the other

`contains_properly` still tests if "all points of A is contained by B", what you need is a `contains_any` that test if any of A is contained by B.